src/NLog/Internal/PropertyHelper.cs

```diff
@@ -421,28 +433,25 @@ private static object ExtractHashSetComparer(object obj, PropertyInfo propInfo)
 
         private static bool TryTypeConverterConversion(Type type, string value, out object newValue)
         {
-#if !SILVERLIGHT && !NETSTANDARD1_3
-            var converter = TypeDescriptor.GetConverter(type);
-            if (converter.CanConvertFrom(typeof(string)))
-            {
-                newValue = converter.ConvertFromInvariantString(value);
-                return true;
-            }
-#else
-            if (type == typeof(LineEndingMode))
+            try
             {
-                newValue = LineEndingMode.FromString(value);
-                return true;
+#if !SILVERLIGHT && !NETSTANDARD1_3
+                var converter = TypeDescriptor.GetConverter(type);
```

Comment:
we could unit test this with NSubstitute and a wrapper arround `TypeDescriptor`. what do you think?